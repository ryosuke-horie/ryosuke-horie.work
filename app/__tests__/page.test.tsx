import { render, screen, within } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Home from '../page';

describe('Home', () => {
    // スナップショットテスト
    it('renders homepage unchanged', () => {
        const { container } = render(<Home />);
        expect(container).toMatchSnapshot();
    });

    // 重要なコンテンツの表示確認
    it('renders main sections', () => {
        render(<Home />);
        const nav = screen.getByRole('navigation');
        // ナビゲーション内の特定テキストを検証
        expect(within(nav).getByText(/^WEBエンジニア \| 堀江 遼佑$/)).toBeInTheDocument();

        // それ以外は全体からでもOK
        expect(screen.getByText(/サービス内容/)).toBeInTheDocument();
        expect(screen.getByText(/実績・事例/)).toBeInTheDocument();
        // 「プロフィール」も同様に範囲を絞るなど
        const profileSection = screen.getByRole('heading', { name: 'プロフィール' });
        expect(profileSection).toBeInTheDocument();
    });

    // ナビゲーションリンクの確認
    it('contains all navigation links', () => {
        render(<Home />);
        // ナビゲーション領域を取得
        const nav = screen.getByRole('navigation');
        const links = ['ホーム', 'サービス', '実績', 'プロフィール', '料金', 'お問い合わせ'];
        links.forEach((link) => {
            // ナビゲーション内のみで検索
            expect(within(nav).getByText(link)).toBeInTheDocument();
        });
    });

    // お問い合わせフォームの存在確認
    it('renders contact form with required fields', () => {
        render(<Home />);
        expect(screen.getByPlaceholderText(/山田 太郎/)).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/your@email.com/)).toBeInTheDocument();
        expect(screen.getByPlaceholderText(/ご相談内容をご記入ください/)).toBeInTheDocument();
    });
});
