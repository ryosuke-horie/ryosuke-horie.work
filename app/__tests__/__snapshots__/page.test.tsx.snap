// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders homepage unchanged 1`] = `
<div>
  <div
    class="min-h-screen bg-white"
    data-oid="zfvm6p5"
  >
    <nav
      class="fixed w-full z-50 bg-white shadow-md"
      data-oid="4ttnykc"
    >
      <div
        class="max-w-7xl mx-auto flex justify-between items-center p-6"
        data-oid="yz6y903"
      >
        <div
          class="text-xl font-bold text-gray-800"
          data-oid="j4ne60o"
        >
          WEBエンジニア | 堀江 遼佑
        </div>
        <div
          class="space-x-4"
          data-oid="4ge:q5l"
        >
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="b92.53o"
            href="#hero"
          >
            ホーム
          </a>
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="wf0c.lm"
            href="#services"
          >
            サービス
          </a>
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="dol8hch"
            href="#portfolio"
          >
            実績
          </a>
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="48wurwb"
            href="#profile"
          >
            プロフィール
          </a>
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="95y33a0"
            href="#pricing"
          >
            料金
          </a>
          <a
            class="text-gray-800 hover:text-blue-600"
            data-oid="dxwzeav"
            href="#contact"
          >
            お問い合わせ
          </a>
        </div>
      </div>
    </nav>
    <main
      class="pt-20"
      data-oid="exveen8"
    >
      <section
        class="min-h-screen flex items-center justify-center relative bg-white overflow-hidden"
        data-oid="i0o51xu"
        id="hero"
      >
        <div
          class="absolute inset-0"
          data-oid="fizl5w3"
        >
          <div
            class="absolute top-0 left-1/2 transform -translate-x-1/2 w-[1000px] h-[1000px] bg-gradient-to-br from-blue-500 via-purple-500 to-orange-400 blur-3xl opacity-40 animate-pulse"
            data-oid="bnky11_"
          />
        </div>
        <div
          class="relative z-10 text-center px-4"
          data-oid="8ssuxla"
        >
          <h1
            class="text-4xl md:text-5xl font-bold text-gray-900 mb-6"
            data-oid="uvdokte"
          >
            格闘技ジムの現場を知る
            <br
              data-oid="xgj2_vx"
            />
            WEBエンジニアが、
            <br
              data-oid="zqzcjz_"
            />
            伴走型の制作・開発で
            <br
              data-oid="1s6sjft"
            />
            あなたの道場運営を支えます
          </h1>
          <p
            class="text-lg md:text-xl text-gray-700 mb-8"
            data-oid="qq2m87m"
          >
            ITが苦手なオーナーにも安心していただける、寄り添い型のサポート体制を実現。
          </p>
          <a
            class="inline-block px-8 py-4 bg-blue-600 text-white font-semibold rounded-lg shadow hover:bg-blue-500 transition-colors"
            data-oid="y2purga"
            href="#contact"
          >
            まずはご相談ください
          </a>
        </div>
      </section>
      <section
        class="py-20 bg-gray-50"
        data-oid=".w-43qw"
        id="services"
      >
        <div
          class="max-w-7xl mx-auto px-6"
          data-oid="l74umu:"
        >
          <h2
            class="text-3xl font-bold text-center text-gray-900 mb-4"
            data-oid="8x_jol_"
          >
            サービス内容
          </h2>
          <p
            class="text-center text-gray-700 mb-12"
            data-oid="42nrs:2"
          >
            伴走型の開発体制で、ジム運営を全面サポート。
            <br
              data-oid="n9rmo-x"
            />
            ホームページ制作・リプレイス、予約管理システム、継続的な保守・運用支援を提供します。
          </p>
          <div
            class="grid grid-cols-1 md:grid-cols-3 gap-8"
            data-oid="_:bhhw9"
          >
            <div
              class="p-6 rounded-xl bg-white shadow-md hover:shadow-xl transition-shadow"
              data-oid="gaxvsx8"
            >
              <div
                class="h-12 w-12 bg-blue-100 rounded-full flex items-center justify-center mb-4"
                data-oid="qbj39rz"
              >
                <svg
                  class="w-6 h-6 text-blue-600"
                  data-oid="cw6w.x:"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 7h18M3 12h18M3 17h18"
                    data-oid="kbpkbmi"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold text-gray-900 mb-2"
                data-oid="u:p509k"
              >
                ホームページ制作
              </h3>
              <p
                class="text-gray-700"
                data-oid="wx5r9qt"
              >
                オリジナルデザインのホームページ制作や既存ホームページのリプレイスを行います。
                <br
                  data-oid=":su6maz"
                />
                集客力の向上のため、内部SEO対策、MEO対策も実施いたします。
              </p>
            </div>
            <div
              class="p-6 rounded-xl bg-white shadow-md hover:shadow-xl transition-shadow"
              data-oid="cpkv5se"
            >
              <div
                class="h-12 w-12 bg-blue-100 rounded-full flex items-center justify-center mb-4"
                data-oid="aiobeco"
              >
                <svg
                  class="w-6 h-6 text-blue-600"
                  data-oid="-xe9-qr"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8 7V3M16 7V3M4 11h16M4 15h16M4 19h16"
                    data-oid="eircrx2"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold text-gray-900 mb-2"
                data-oid="_y5qwdu"
              >
                システム開発
              </h3>
              <p
                class="text-gray-700"
                data-oid="fnb7d86"
              >
                予約管理システムや会員管理システムなど、ジム運営を効率化するシステムを開発いたします。
              </p>
            </div>
            <div
              class="p-6 rounded-xl bg-white shadow-md hover:shadow-xl transition-shadow"
              data-oid="v6lr-to"
            >
              <div
                class="h-12 w-12 bg-blue-100 rounded-full flex items-center justify-center mb-4"
                data-oid="nc4si--"
              >
                <svg
                  class="w-6 h-6 text-blue-600"
                  data-oid="o.ylkm8"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M13 10V3L4 14h7v7l9-11h-7z"
                    data-oid="3sezk5k"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <h3
                class="text-xl font-semibold text-gray-900 mb-2"
                data-oid="tal1yl8"
              >
                継続的な保守・運用サポート
              </h3>
              <p
                class="text-gray-700"
                data-oid="iyikb43"
              >
                一度作って終わりではなく、伴走型での運用支援により、常に最適な状態を維持いたします。
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="py-20"
        data-oid="tkyijlf"
        id="portfolio"
      >
        <div
          class="max-w-7xl mx-auto px-6"
          data-oid="f207dpp"
        >
          <h2
            class="text-3xl font-bold text-center text-gray-900 mb-4"
            data-oid=".lr0grf"
          >
            実績・事例
          </h2>
          <p
            class="text-center text-gray-700 mb-12"
            data-oid="vjrcj94"
          >
            過去のプロジェクトを通じて、道場運営の課題を解決した事例をご紹介
          </p>
          <div
            class="grid grid-cols-1 md:grid-cols-2 gap-8"
            data-oid="g8o6._r"
          >
            <div
              class="group relative bg-white rounded-xl overflow-hidden border border-gray-200 shadow-md hover:shadow-xl transition-shadow"
              data-oid="new-card"
            >
              <a
                class="block"
                data-oid="ome.9a4"
                href="https://timetable-hideskick.net/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <div
                  class="aspect-video overflow-hidden"
                  data-oid="new-image"
                >
                  &lt;img src="/images/hides-timetable.webp" alt="体験予約受付システム hides timetable" width="600" height="400" className="object-cover w-full h-full transform group-hover:scale-105 transition-transform duration-300" data-oid="ghq_7ts" /&gt;
                </div>
                <div
                  class="p-6"
                  data-oid="new-info"
                >
                  <h3
                    class="text-xl font-semibold text-gray-900 mb-2"
                    data-oid="new-title"
                  >
                    体験予約受付システム
                    <span
                      class="text-sm text-gray-500"
                      data-oid="new-client"
                    >
                      Hide's Kick様
                    </span>
                  </h3>
                  <p
                    class="text-gray-700"
                    data-oid="new-desc"
                  >
                    オンラインでの体験予約受付システムを開発。タイムテーブルに模した予約フォームを実現。
                  </p>
                </div>
              </a>
            </div>
            <div
              class="group relative bg-white rounded-xl overflow-hidden border border-gray-200 shadow-md hover:shadow-xl transition-shadow"
              data-oid="5h1_cq."
            >
              <div
                class="aspect-video overflow-hidden"
                data-oid="ypppr6h"
              >
                <svg
                  class="object-cover w-full h-full transform group-hover:scale-105 transition-transform duration-300"
                  data-oid="u3gfkoi"
                  viewBox="0 0 600 400"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    data-oid="k_0m.s9"
                    fill="#cccccc"
                    height="400"
                    width="600"
                  />
                  <text
                    data-oid="o.20:cm"
                    dominant-baseline="middle"
                    fill="#666666"
                    font-size="24"
                    text-anchor="middle"
                    x="50%"
                    y="50%"
                  >
                    Sample Image
                  </text>
                </svg>
              </div>
              <div
                class="p-6"
                data-oid="gakiip0"
              >
                <h3
                  class="text-xl font-semibold text-gray-900 mb-2"
                  data-oid="gmit9md"
                >
                  ホームページ制作
                  <span
                    class="text-sm text-gray-500"
                    data-oid="z9z7z7_"
                  >
                    HK Studio様
                  </span>
                </h3>
                <p
                  class="text-gray-700"
                  data-oid="g9t-yp8"
                >
                  オリジナルデザインのホームページ制作。パーソナルジムの雰囲気を活かしたデザインを実現。
                </p>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        class="py-20 bg-gray-50"
        data-oid="ik1olm1"
        id="profile"
      >
        <div
          class="max-w-4xl mx-auto px-6 text-center"
          data-oid="y33jtgv"
        >
          <h2
            class="text-3xl font-bold text-gray-900 mb-4"
            data-oid="ays.314"
          >
            プロフィール
          </h2>
          <div
            class="flex flex-col md:flex-row items-center justify-center"
            data-oid="v4hgoug"
          >
            &lt;img src="/images/my_image.webp" alt="WEBエンジニア 堀江 遼佑のプロフィール画像" width="200" height="200" className="rounded-full mb-6 md:mb-0 md:mr-8" data-oid=":o9o8e-" /&gt;
            <div
              class="text-left"
              data-oid="-sp7wae"
            >
              <p
                class="text-xl text-gray-900 font-semibold"
                data-oid="7mr.wyo"
              >
                WEBエンジニア × 格闘技ジムスタッフ
              </p>
              <p
                class="text-gray-700 mt-4"
                data-oid="._7hm58"
              >
                WEBエンジニアとしての開発経験、さらに格闘技ジムでの現場経験を活かし、ITが苦手な小規模ジムや個人事業オーナーに寄り添ったサポートを提供しています。
              </p>
            </div>
          </div>
        </div>
      </section>
      <section
        class="py-20 bg-gray-50"
        data-oid="nd8xple"
        id="pricing"
      >
        <div
          class="max-w-5xl mx-auto px-6"
          data-oid="jwolf:t"
        >
          <h2
            class="text-3xl font-bold text-center text-gray-900 mb-4"
            data-oid="gd7k2yh"
          >
            料金・プラン
          </h2>
          <p
            class="text-center text-gray-700 mb-12"
            data-oid="udt-l96"
          >
            ニーズに合わせてご相談させていただきます。以下は目安となります。
          </p>
          <div
            class="grid grid-cols-1 md:grid-cols-2 gap-8"
            data-oid="dp..yb3"
          >
            <div
              class="p-6 rounded-xl bg-white shadow-md"
              data-oid=".pjdfd7"
            >
              <h3
                class="text-xl font-semibold text-gray-900 mb-2"
                data-oid="4oyjf_j"
              >
                ホームページ制作
              </h3>
              <p
                class="text-gray-700 mb-4"
                data-oid="tj-.n2m"
              >
                初期費用：¥30,000～
              </p>
              <p
                class="text-gray-700"
                data-oid=".lyu2:z"
              >
                内部SEO対策とMEO対策を実施し、集客力の向上をサポートします。
              </p>
            </div>
            <div
              class="p-6 rounded-xl bg-white shadow-md"
              data-oid="tlyh15l"
            >
              <h3
                class="text-xl font-semibold text-gray-900 mb-2"
                data-oid="6yelke3"
              >
                システム開発
              </h3>
              <p
                class="text-gray-700 mb-4"
                data-oid="ff67gcf"
              >
                初期費用：¥100,000～
              </p>
              <p
                class="text-gray-700"
                data-oid=".hmr5s3"
              >
                ご要望に合わせたシステムをスクラッチ開発いたします。プロジェクト規模に合わせて柔軟にお見積りいたします。
              </p>
            </div>
          </div>
          <div
            class="mt-8 text-center"
            data-oid="kgue9mh"
          >
            <p
              class="text-gray-700"
              data-oid="-r-tus8"
            >
              【導入の流れ】 ヒアリング → 要件定義 → デザイン・開発 → 納品 → 運用・継続サポート
            </p>
          </div>
        </div>
      </section>
      <section
        class="py-20 bg-gray-100"
        data-oid="9erkxzg"
        id="contact"
      >
        <div
          class="max-w-3xl mx-auto px-6"
          data-oid="l6vuqj7"
        >
          <h2
            class="text-3xl font-bold text-center text-gray-900 mb-4"
            data-oid="4bt1nti"
          >
            お問い合わせ
          </h2>
          <p
            class="text-center text-gray-700 mb-12"
            data-oid="dviv7j8"
          >
            お気軽にご連絡ください。ご相談内容に合わせた最適なご提案をいたします。
          </p>
          <form
            class="space-y-6"
            data-oid="jkzj_w4"
          >
            <div
              data-oid="b2ueipz"
            >
              <label
                class="block text-gray-800 mb-2"
                data-oid=".l_-3yb"
              >
                お名前
              </label>
              <input
                class="w-full px-4 py-3 rounded-lg bg-white border border-gray-300"
                data-oid="xpnxq.f"
                placeholder="山田 太郎"
                type="text"
              />
            </div>
            <div
              data-oid="3t7-ifu"
            >
              <label
                class="block text-gray-800 mb-2"
                data-oid="m3c24cq"
              >
                メールアドレス
              </label>
              <input
                class="w-full px-4 py-3 rounded-lg bg-white border border-gray-300"
                data-oid="_:9cs18"
                placeholder="your@email.com"
                type="email"
              />
            </div>
            <div
              data-oid="3_5s85_"
            >
              <label
                class="block text-gray-800 mb-2"
                data-oid="eodw3os"
              >
                お問い合わせ内容
              </label>
              <textarea
                class="w-full px-4 py-3 rounded-lg bg-white border border-gray-300"
                data-oid="a3e8on:"
                placeholder="ご相談内容をご記入ください"
              />
            </div>
            <div
              class="text-center pt-4"
              data-oid="43ymkzd"
            >
              <button
                class="px-8 py-4 bg-blue-600 text-white rounded-lg hover:bg-blue-500 transition-colors"
                data-oid="bwqhrtw"
                type="submit"
              >
                送信する
              </button>
            </div>
          </form>
        </div>
      </section>
    </main>
    <footer
      class="bg-gray-200 text-gray-700 py-8"
      data-oid=":k1zf7t"
    >
      <div
        class="max-w-7xl mx-auto px-6 text-center"
        data-oid="9wi-u53"
      >
        <p
          data-oid="yxl2q8j"
        >
          © 2025 ryosuke horie . All rights reserved.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
